src/Microsoft.DotNet.Darc/src/DarcLib/VirtualMonoRepo/VmrCloakedFileScanner.cs

```diff
@@ -43,19 +45,59 @@ public class VmrCloakedFileScanner : VmrScanner
 
         foreach (var exclude in sourceMapping.Exclude)
         {
-            args.Add(GetExclusionFilter(baseExcludePath / exclude));
+            args.Add(GetCloakedFileFilter(baseExcludePath / exclude));
         }
 
         args.AddRange(await GetExclusionFilters(sourceMapping.Name, baselineFilePath));
 
+        return await ScanAndParseResult(args.ToArray(), sourceMapping.Name, cancellationToken);
+
+    }
+
+    protected override string ScanType { get; } = "cloaked";
+    private string GetCloakedFileFilter(string file) => $":(attr:!{VmrInfo.KeepAttribute}){file}";
+
+    protected override async Task<IEnumerable<string>> ScanBaseRepository(string baselineFilePath, CancellationToken cancellationToken)
+    {
+        cancellationToken.ThrowIfCancellationRequested();
+
+        var args = new List<string>
+        {
+            "diff",
+            "--name-only",
+            Constants.EmptyGitObject,
+            $":(exclude){VmrInfo.SourcesDir}"
+        };
+
+        foreach (var exclude in GetGlobalExcludeFilters())
+        {
+            args.Add(GetCloakedFileFilter(_vmrInfo.VmrPath / exclude));
+        }
+
+        args.AddRange(await GetExclusionFilters(null, baselineFilePath));
+
+        return await ScanAndParseResult(args.ToArray(), "base VMR", cancellationToken);
+    }
+
+    private async Task<IEnumerable<string>> ScanAndParseResult(string[] args, string repoName, CancellationToken cancellationToken)
+    {
         var ret = await _processManager.ExecuteGit(_vmrInfo.VmrPath, args.ToArray(), cancellationToken);
 
-        ret.ThrowIfFailed($"Failed to scan the {sourceMapping.Name} repository");
+        ret.ThrowIfFailed($"Failed to scan the {repoName} repository");
 
         return ret.StandardOutput
             .Split(Environment.NewLine, StringSplitOptions.RemoveEmptyEntries);
     }
 
-    protected override string ScanType { get; } = "cloaked";
-    private string GetExclusionFilter(string file) => $":(attr:!{VmrInfo.KeepAttribute}){file}";
+    private IEnumerable<string> GetGlobalExcludeFilters()
```

Comment:
This won't work - a mapping can override the defaults and have a completely different set of filters so you'd get an empty result here.

What are you trying to achieve with this?

Reply:
Ahh ok, I was trying to get the non repo specific cloaked file filters to apply them on the base repo, by creating an intersection between all of the repo specific ones.
I will add the global Excludes somewhere as a property when we're loading them from the file

Reply:
@dkurepa but I don't think we need to exclude files outside of `src/` though. We just need to know, when we get a binary there

Reply:
Ah ok, I guess it makes sense not to scan the base VMR repo for the cloaked files

Reply:
Yup, just binaries there are fine